addons/pos_self_order/controllers/controllers.py

```diff
@@ -0,0 +1,263 @@
+        if not find_any_pos_with_qr_code_menu_sudo():
+            raise werkzeug.exceptions.NotFound()
+
+        product_sudo = request.env["product.product"].sudo().browse(product_id)
+        if not product_sudo.available_in_pos:
+            raise werkzeug.exceptions.NotFound()
+
+        if size == 1920 and product_sudo.image_1920:
+            return (
+                request.env["ir.binary"]
+                ._get_image_stream_from(product_sudo, field_name="image_1920")
+                .get_response()
+            )
+        return (
+            request.env["ir.binary"]
+            ._get_image_stream_from(product_sudo, field_name="image_128")
+            .get_response()
+        )
+
+    @http.route("/menu/get-bg-image", methods=["GET"], type="http", auth="public")
+    def pos_self_order_get_bg_image(self, pos_id: int):
+        """
+        Gets the background image for this self order
+        :param pos_id: the id of the POS
+        :type pos_id: int
+        :return: the bg image
+        :rtype: binary
+        """,
+        pos_sudo = find_pos_with_id_sudo(pos_id)
+        if not pos_sudo.self_order_image:
+            raise werkzeug.exceptions.NotFound()
+        return (
+            request.env["ir.binary"]
+            ._get_image_stream_from(pos_sudo, field_name="self_order_image")
+            .get_response()
+        )
+
+
+def find_pos_sudo(pos_sudo=None, pos_id: Optional[int] = None):
+    """
+    Returns the pos_sudo object, if it exists, otherwise it raises an error
+    :param pos_sudo: the pos_sudo object
+    :type pos_sudo: pos.config
+    :param pos_id: the id of the POS
+    :type pos_id: int
+    :return: the pos_sudo object
+    :rtype: pos.config
+    """
+    if pos_sudo:
+        return pos_sudo
+    if pos_id:
+        return find_pos_with_id_sudo(pos_id)
+    return find_any_pos_with_qr_code_menu_sudo()
+
+
+def find_any_pos_with_qr_code_menu_sudo():
+    """
+    This function finds a POS that allows the QR code menu, and returns it
+    or raises an error if there is no POS that allows the QR code menu
+    :return: the pos config object
+    :rtype: pos.config
+    """
+    pos_sudo = (
+        request.env["pos.config"]
+        .sudo()
+        .search([("self_order_view_mode", "=", True)], limit=1)
+    )
+    if not pos_sudo:
+        raise werkzeug.exceptions.NotFound()
+    return pos_sudo
+
+
+def find_pos_with_id_sudo(pos_id: int):
+    """
+    This function checks that the pos_id exists, and that the pos is configured to allow the menu to be viewed online
+    :param pos_id: the id of the POS
+    :type pos_id: int
+    :return: the pos config object, if the pos_id is valid
+    :rtype: pos.config
+    """
+    pos_sudo = request.env["pos.config"].sudo().search([("id", "=", pos_id)])
+    if not pos_sudo or not pos_sudo.self_order_view_mode:
+        raise werkzeug.exceptions.NotFound()
+    return pos_sudo
+
+
+def get_self_order_config(pos_sudo) -> Dict[str, str]:
```

Comment:
We can remove the type because it's wrong. Better not have it than being confusing.

Reply:
what if i change it to this `def get_self_order_config(pos_sudo) -> Dict[str]:` ?
The idea is to have a dictionary where the keys are strings